<launch>
    <!-- base controller -->
    <node pkg="base_controller" type="base_controller" name="base_controller" output="screen">
        <param name="motor_max_acc" value="200.000" />
        <param name="motor_max_vel" value="37.699111843" />
        <param name="motor_vel_coeff" value="33.333333333" />
        <param name="robot_radius" value="0.150" />
        <param name="invert_x" value="true" />
        <param name="invert_y" value="false" />
        <param name="invert_z" value="true" />
    </node>

    <!-- joy teleop -->
    <node pkg="base_controller" type="base_teleop_joy" name="base_teleop_joy" output="screen">
        <param name="max_lin" value="1.0" />
        <param name="max_lin_turbo" value="1.0" />
        <param name="max_ang" value="3.141592" />
        <param name="max_ang_turbo" value="3.141592" />
    </node>

    <node pkg="base_controller" type="shutdown_controller_joy" name="sd_controller" />

    <!-- joy node -->
    <node respawn="true" pkg="joy" type="joy_node" name="joy_node">
        <param name="dev" type="string" value="/dev/input/js0" />
        <param name="deadzone" value="0.12" />
    </node>

    <!-- base itself -->
    <node pkg="rosserial_server" type="serial_node" name="base">
        <rosparam>
            port: /dev/ttyUSB0
            baud: 921600
        </rosparam>
    </node>

    <node pkg="rosserial_python" type="message_info_service.py" name="rosserial_message_info" />
</launch>
